core/trino-main/src/main/java/io/trino/connector/system/TrinoMaterializedViewAdapter.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.connector.system;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.SchemaTableName;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+
+public class TrinoMaterializedViewAdapter
+        implements MaterializedViewRowAdapter
+{
+    @Override
+    public ConnectorTableMetadata getTableLayout()
+    {
+        return new ConnectorTableMetadata(getTableName(), getTableColumns());
+    }
+
+    @Override
+    public SchemaTableName getTableName()
+    {
+        return new SchemaTableName("metadata", "materialized_views");
+    }
+
+    @Override
+    public List<ColumnMetadata> getTableColumns()
+    {
+        return ImmutableList.of(
+                new ColumnMetadata("catalog_name", createUnboundedVarcharType()),
+                new ColumnMetadata("schema_name", createUnboundedVarcharType()),
+                new ColumnMetadata("name", createUnboundedVarcharType()),
+                new ColumnMetadata("storage_catalog", createUnboundedVarcharType()),
+                new ColumnMetadata("storage_schema", createUnboundedVarcharType()),
+                new ColumnMetadata("storage_table", createUnboundedVarcharType()),
+                new ColumnMetadata("is_fresh", BOOLEAN),
+                new ColumnMetadata("owner", createUnboundedVarcharType()),
+                new ColumnMetadata("comment", createUnboundedVarcharType()),
+                new ColumnMetadata("text", createUnboundedVarcharType()));
```

Comment:
text -> materialized_view_definition

Reply:
Do we need to prefix every column with materialized_view? The table column names are very long and the table does not fit on a single monitor anymore.

Perhaps we could just have `materialized_view_name`, `definition`,  `catalog_name`, `schema_name`, etc? This would look consistent with `system.metadata.table_comments` which has columns: `catalog_name`, `schema_name`, `table_name`.

Reply:
I was trying to make it somewhat similar to InformationSchemaTable where we use table_catalog, table_schema, table_name, view_definition but it's not really necessary.
We can keep shorter names: name, definition, catalog_name, schema_name

Reply:
That makes sense. Maybe there should be a task to make tables in `system.metadata` and `information_schema` consistent, since I noticed the different naming schemes.